### ISSUE TITLE
Sanity Checks for Saliency Maps

### ISSUE THREAD

**Issue by Tonnonssi** [Tonnonssi/xRL#5]
## 🥅 목적
Sanity Checks for Saliency Maps 논문 리뷰  

## 🤩 공부할 내용
- 논문 정리 
- 각 xAI 방법론 이해  
- 논문 의의

## 📄 참고 자료 
[Sanity Checks for Saliency Maps](https://arxiv.org/abs/1810.03292)

**Comment by Tonnonssi** [comment]
## 서론 
Saliency : 학습한 모델 기반 예측을 통해 피쳐를 하이라이트하는 방법론  
😀 : 직관적인 이해가 가능함, 많은 xAI 방법론의 접근 방식   
😟 : 사람들은 Saliency를 보며 어떠한 패턴을 찾아내려고 노력하기에, noise를 해석해 버리는 **확증 편향** 문제가 발생  

> 🌟 Saliency 기반 방법론들이 과연 진짜 의미가 있는 방법인지 진위여부를 가림  

## 논문이 제시하는 방법론 
### 1. The model parameter randomization test
1-1. cascading 방법론 : 누적해서 파라미터를 섞음 
1-2. single layer 방법론 : 독립된 하나의 레이어만 섞음  

[ 해석 방법 ]  
' 학습된 모델 vs 학습되지 않은 랜덤한 모델 ' 의 Saliency를 비교한다. 신경망의 파라미터에 종속됨. 
- 결과가 다르다 : 학습한 특정 부분의 파라미터가 Saliency에 영향을 주었다.   
- 결과가 유사하다 : 학습한 특정 부분의 파라미터가 Saliency에 영향을 주지 않는다.  


### 2. The data randomization test  
' 라벨을 제대로 달아서 학습 vs 라벨을 다 섞어서 학습 ' 의 원모델과 섞은 모델의 Saliency 상관도를 분석한다.  
- 상관이 높다 : 실제 라벨(정답)에 종속된다.  
- 상관이 낮다 : 실제 정답에 종속되지 않는다.  

## 사용하는 xAI 방법론 
1. Gradient⊙input
2. Integrated Gradients (IG) 
3. Guided Backpropagation (GBP)
4. Guided GradCAM
5. SmoothGrad (SG) 

## 분석 결과   
1. **파라미터 랜덤화 - cascading** 
<img width="655" height="361" alt="Image" src="https://github.com/user-attachments/assets/86aaa2eb-551c-46c2-9ed8-39dd9ff7e035" />

gradient⊙input, IG 베이스 모델과 유사하다. -> 썩 좋지 않네요... 하지만 눈으로 보는 건 편향이 있을 수 있다. 데이터 기반 분석은 스피어만 상관 계수 등을 사용해 수치화한다.   

- Spearman Rank correlation   
<img width="671" height="402" alt="Image" src="https://github.com/user-attachments/assets/b7b49e70-3255-4b89-823a-5cba6450f1a9" />  

- SSIM and HOGs similarity   
<img width="647" height="274" alt="Image" src="https://github.com/user-attachments/assets/4301e779-033e-4e1b-930b-01f3ad0df4ee" />

| 지표 | 설명 | 
| - | - | 
| SSIM(Structural Similarity Index) | 두 이미지 간의 구조적 유사도를 측정한다. 1에 가까울수록 동일한 것으로 해석 | 
| Spearman Rank Correlation | 픽셀 중요도를 값의 순위 상관 계수로 측정, 정확한 수치화된 차이가 아니라, 열등 우등 관계만 파악하기 위함이다.  | 
| HOG (Histogram of Oriented Gradients) Similarity | 두 맵의 에지 방향 분포 간의 상관관계를 측정한다. | 

2.  **파라미터 랜덤화 - independent** 

<img width="633" height="341" alt="Image" src="https://github.com/user-attachments/assets/257b382a-e524-4896-b151-817c962f29a7" />.  

Guided Backprop, Guided GradCAM은 초반 레이어에만 민감하고, 뒤에 애들은 그닥 의미없다.  

3. **데이터 랜덤화** 

지도학습에서 라벨을 섞는다는 건 진짜 말도 안되는 똥 모델이라는 뜻. 이런 모델과 비교했을 때 비슷하다는 건 진짜 의미없다는 거랑 동일하다.  
제대로 학습한 모델은 train set에 대해 약 95%의 acc 

<img width="662" height="305" alt="Image" src="https://github.com/user-attachments/assets/f2492a7a-a88a-49dd-9af9-f9fec47cc965" />  

- GradCam, Gradient는 상관도가 낮다. 특히 GradCam이 많이 낮다.  
- Gradient-Input & IG는 상관도가 정말 높다.  

### Edge Detector 
<img width="616" height="261" alt="Image" src="https://github.com/user-attachments/assets/75705969-d4de-4bd6-8474-1e83b019ab73" />  

가장자리 검출법은 Saliency와 거의 유사하다. 하지만 이게 설명가능하다고 말할 수는 없다. 근거가 부족하고, 이걸 설명하는 최소한의 함수가 존재하지 않기 때문이다.  

## 결론 
시각적 분석은 모델과 데이터 생성 과정에서 민감도가 떨어진다. Gradient, GradCAM 계열은 좀 괜찮지만, Guided BackProp은 그냥 엣지 디텍터 이상의 성능을 보이지 못한다. 따라서 xAI의 툴로써 사용하는 것이 부적절하다. 

## 시사점 
이 연구는 "보기에 좋은 설명이 좋은 설명이다"라는 순진한 가정을 무너뜨리고, 설명 가능성(Explainability)을 과학적으로 검증해야 한다는 엄격한 기준을 제시했다. 이전에는 시각적으로 좋은 GBP가 선호되었고, 노이즈가 많은 Gradient, GradCAM은 기피 대상이었다. 하지만 이 연구는 GBP가 엣지 디텍터 이상의 성능을 내지 못하며, Gradient, GradCAM의 이론적 해석 가능성의 완전함을 시사해 xAI의 접근 방식을 흔들어 놓았다.
